### ISSUE TITLE
CUDA tests not working anymore

### ISSUE THREAD

**Issue by bramton** [opencv/opencv#7993]
##### System information (version)
- OpenCV => 3.2
- Operating System / Platform => Linux 64 Bit
- Compiler => gcc

##### Detailed description
Compiling the latest source with BUILD_TESTS enabled produces the testing binaries. Running CUDA related tests have no output.

```.sh
./opencv_test_cudaarithm
CTEST_FULL_OUTPUT
OpenCV version: 3.2.0-dev
OpenCV VCS version: 3.2.0-29-gec5038f-dirty
Build type: release
CPU features: mmx sse sse2 sse3

Available options besides google test option: 
Usage: opencv_test_cudaarithm [params] 

	--cuda_device (value:-1)
		CUDA device on which tests will be executed (-1 means all devices)
	-h, --help (value:false)
		Print help info

Run tests on all supported CUDA devices 

[----------]
[ GPU INFO ] 	Run on OS Linux x64.
[----------]
*** CUDA Device Query (Runtime API) version (CUDART static linking) *** 

Device count: 1

Device 0: "GeForce GTX 960"
  CUDA Driver Version / Runtime Version          8.0 / 8.0
  CUDA Capability Major/Minor version number:    5.2
  Total amount of global memory:                 4038 MBytes (4233691136 bytes)
  GPU Clock Speed:                               1.25 GHz
  Max Texture Dimension Size (x,y,z)             1D=(65536), 2D=(65536,65536), 3D=(4096,4096,4096)
  Max Layered Texture Size (dim) x layers        1D=(16384) x 2048, 2D=(16384,16384) x 2048
  Total amount of constant memory:               65536 bytes
  Total amount of shared memory per block:       49152 bytes
  Total number of registers available per block: 65536
  Warp size:                                     32
  Maximum number of threads per block:           1024
  Maximum sizes of each dimension of a block:    1024 x 1024 x 64
  Maximum sizes of each dimension of a grid:     2147483647 x 65535 x 65535
  Maximum memory pitch:                          2147483647 bytes
  Texture alignment:                             512 bytes
  Concurrent copy and execution:                 Yes with 2 copy engine(s)
  Run time limit on kernels:                     No
  Integrated GPU sharing Host Memory:            No
  Support host page-locked memory mapping:       Yes
  Concurrent kernel execution:                   Yes
  Alignment requirement for Surfaces:            Yes
  Device has ECC support enabled:                No
  Device is using TCC driver mode:               No
  Device supports Unified Addressing (UVA):      Yes
  Device PCI Bus ID / PCI location ID:           1 / 0
  Compute Mode:
      Default (multiple host threads can use ::cudaSetDevice() with device simultaneously) 

deviceQuery, CUDA Driver = CUDART, CUDA Driver Version  = 8.0, CUDA Runtime Version = 8.0, NumDevs = 1

[==========] Running 0 tests from 0 test cases.
[==========] 0 tests from 0 test cases ran. (0 ms total)
[  PASSED  ] 0 tests.
```
Other tests do work, such as `opencv_test_core`. If I checkout the OpenCV 3.1.0 branch the CUDA related tests work again. Perhaps something went wrong with the gtest upgrade from 1.7 to 1.8 ?

**Comment by alalek** [comment]
More details are required:
- CMake stdout (from clean build) with information about CUDA settings
- files from build folder: CMakeCache.txt and CMakeVars.txt (as attachment or use "grep" for CUDA keyword)

**Comment by alalek** [comment]
@bramton Could you check fix #7998 please?
